### PR TITLE
Various bug fixes in the network stack

### PR DESCRIPTION
**Pull request by hlef**
This PR clears some of my bugfix TODO list. This addresses a lot of documentation issues, but also actual bugs such as https://github.com/CHERIoT-Platform/network-stack/issues/43.

### CODE REVIEW

File: include/endianness.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// Copyright SCI Semiconductor and CHERIoT Contributors.
+// SPDX-License-Identifier: MIT
```
Comment (davidchisnall) [inline include/endianness.h:2]: Since this is C++-only, it should probably be named .hh.

File: lib/tcpip/network-internal.h [hunk 2]
```diff
@@ -46,7 +46,7 @@ __cheri_compartment("TCPIP") int network_host_resolve(
  *
  * This returns a sealed capability to a socket on success, or null on failure.
  *
- * This should be called only from the NetAPI compartment.
+ * This should be called only from the NetAPI or TCP/IP compartments.
```
Comment (davidchisnall) [inline lib/tcpip/network-internal.h:49]: Why is it called from the TCP/IP compartment? Wouldn’t that use internal APIs?

This contradicts the property enforced in rego:
https://github.com/CHERIoT-Platform/network-stack/blob/473249a6ace45f1d1fe9a1606705355b190327b3/network_stack.rego#L167

Comment (hlef) [reply lib/tcpip/network-internal.h:49]: We use it in `network_socket_udp`, which is just a light wrapper for `network_socket_create_and_bind` with connection type UDP. I'm not sure using an internal API is worth it, the current code makes sense. We should update the rego as well, will update.

**Comment by hlef** [comment]
Sounds like CI is failing because of https://github.com/CHERIoT-Platform/cheriot-rtos/commit/c3c0fb1d4b2d865cb51081c952325f556e25a3f1

**Comment by hlef** [comment]
Comments addressed!

**Comment by hlef** [comment]
> Sounds like CI is failing because of [CHERIoT-Platform/cheriot-rtos@c3c0fb1](https://github.com/CHERIoT-Platform/cheriot-rtos/commit/c3c0fb1d4b2d865cb51081c952325f556e25a3f1)

Should be fixed now.

**Pull request merged**